src/lib/api/clients/impls/data-api-http-client.ts

```diff
@@ -0,0 +1,374 @@
+    this.emissionStrategy = opts.emissionStrategy(opts.logger.internal, this._baseUrl);
+    this.bigNumHack = opts.bigNumHack;
+    this.#baseOpts = opts;
+  }
+
+  public forTableSlashCollectionOrWhateverWeWouldCallTheUnionOfTheseTypes(tSlashC: Collection | Table<SomeRow>, opts: CollectionOptions | TableOptions | undefined, bigNumHack: BigNumberHack): DataAPIHttpClient {
+    return new DataAPIHttpClient({
+      ...this.#baseOpts,
+      emissionStrategy: EmissionStrategy.Normal,
+      keyspace: { ref: tSlashC.keyspace },
+      logger: tSlashC,
+      additionalHeaders: HeadersProvider.opts.monoid.concat([
+        this.#baseOpts.additionalHeaders,
+        HeadersProvider.opts.fromStr(EmbeddingAPIKeyHeaderProvider).parse(opts?.embeddingApiKey),
+        HeadersProvider.opts.fromStr(RerankingAPIKeyHeaderProvider).parse(opts?.rerankingApiKey),
+      ]),
+      timeoutDefaults: {
+        ...this.#baseOpts.timeoutDefaults,
+        ...opts?.timeoutDefaults,
+      },
+      collection: tSlashC instanceof Collection ? tSlashC.name : undefined,
+      table: tSlashC instanceof Collection ? undefined : tSlashC.name,
+      bigNumHack,
+    });
+  }
+
+  public forDbAdmin(dbAdmin: DbAdmin, opts: ParsedAdminOptions): DataAPIHttpClient<'admin'> {
+    return new DataAPIHttpClient({
+      ...this.#baseOpts,
+      tokenProvider: TokenProvider.opts.concat([opts.adminToken, this.#baseOpts.tokenProvider]),
+      baseUrl: opts?.endpointUrl ?? this.#baseOpts.baseUrl,
+      emissionStrategy: EmissionStrategy.Admin,
+      logger: dbAdmin,
+      timeoutDefaults: {
+        ...this.#baseOpts.timeoutDefaults,
+        ...opts?.timeoutDefaults,
+      },
+      collection: undefined,
+      table: undefined,
+    });
+  }
+
+  public async executeCommand(command: Record<string, any>, opts: ExecuteDataAPICommandOptions<Kind>): Promise<RawDataAPIResponse> {
+    const metadata = this._mkRequestMetadata(opts.timeoutManager, {
+      target: this.target.forRequest(opts),
+      extra: opts.extraLogInfo,
+      command: command,
+    });
+
+    this.emissionStrategy.emitCommandStarted(metadata, opts);
+
+    let clonedData: RawDataAPIResponse | undefined;
+
+    try {
+      const serialized = (opts.bigNumsPresent)
+        ? this.bigNumHack?.parser.stringify(metadata.command)
+        : JSON.stringify(metadata.command);
+
+      const resp = await this._request({
+        url: metadata.target.url,
+        data: serialized,
+        timeoutManager: opts.timeoutManager,
+        method: HttpMethods.Post,
+      });
+
+      if (resp.status >= 400 && resp.status !== 401) {
+        throw new DataAPIHttpError(resp);
+      }
+
+      const data = (resp.body)
+        ? (this.bigNumHack?.parseWithBigNumbers(resp.body))
+          ? this.bigNumHack?.parser.parse(resp.body)
+          : JSON.parse(resp.body)
+        /* c8 ignore next: exceptional case */
+        : {};
+
+      clonedData = structuredClone(data);

```

Comment:
Is this structured clone so that user-registered handlers don't accidentally modify `data`? If so, that's reasonable, just wondering.